forward43/scraper_kickstarter.py

```diff
@@ -14,15 +16,25 @@ def __init__(self, category_ids, num_pages):
         self.category_ids       = category_ids
         self.num_pages          = num_pages
 
+
     def get_url(self, category, page):
         '''
         Generates a URL to scrape from. It's necessary that category and page parameters are not None
         @category : category id
         @page     : page number
         '''
-        random_seed = 'seed=' + str(random.randint(1, 65536))
-        get_params  = self.default_url_params + [f"category_id={category}", f"page={page}", random_seed]
-        return self.base_url + '&'.join(get_params)
+        
+        ## I inserted a for loop over the keywords to create a list of urls to scrape below that include the search terms we want to use
+        url_term_list = []
+        
+        for term in keywords:
+            search_term = 'term=' + term.replace(' ', '+')
+            random_seed = 'seed=' + str(random.randint(1, 65536))
+            get_params  = self.default_url_params + [f"category_id={category}", f"page={page}", random_seed, search_term]   # added a keyword parameter to the url
+            url         = self.base_url + '&'.join(get_params)
+            url_term_list.append(url)
+            
+        return url_term_list
```

Comment:
This is a good idea, however it's implemented in an illogical place: the function you've placed it in is called `get_url` which is singular. Anyone that reads this that hasn't been part of this PR will (rightfully) assume that this function will return exactly one URL, and the docstring will validate that belief. When it gets used they will get a list of URLs.

I'm not too sure if we want to do a search of each combination of `category_id` and `search_term`, but I haven't had enough of a dive into kickstarter data to know what to decide on that front

I'd place this in another function like `get_urls_by_keyword` and extend the scrape script to do both a loop of the original `crawl by category_id` functionality and a loop of `searching by keyword with(or without) category_id`